src/app/store/technical-records/reducers/technical-record-service.reducer.ts

```diff
@@ -497,10 +497,12 @@ function handleClearADRDetails(state: TechnicalRecordServiceState) {
       if (!listStatementApplicable) {
         sanitisedEditingTechRecord = { ...sanitisedEditingTechRecord, ...nulledBatteryListNumber };
       }
-      // If the ADR body type not includes 'battery', null the battery list number even if the listStatementApplicable is true
+      // If the ADR body type not includes 'battery', null all fields related battery list applicable
       const { techRecord_adrDetails_vehicleDetails_type: vehicleDetailsType } = sanitisedEditingTechRecord;
       if (!vehicleDetailsType?.includes('battery')) {
-        sanitisedEditingTechRecord = { ...sanitisedEditingTechRecord, ...nulledBatteryListNumber };
+        sanitisedEditingTechRecord = {
+          ...sanitisedEditingTechRecord, ...nulledBatteryListNumber, techRecord_adrDetails_listStatementApplicable: null,
```

Comment:
would this 'techRecord_adrDetails_listStatementApplicable' be part of the 'nulledBatteryListNumber'? Can be added to that

Reply:
maybe rename nulledBatteryListNumber to nulledBatteryList

Reply:
No, we can't. I used to have the same thought, but if you look at this code snippet below
`       // If battery list applicable is no, null the battery list number
      const { techRecord_adrDetails_listStatementApplicable: listStatementApplicable } = sanitisedEditingTechRecord;
      if (!listStatementApplicable) {
        sanitisedEditingTechRecord = { ...sanitisedEditingTechRecord, ...nulledBatteryListNumber };
      }`
You could find the different logic the `nulledBatteryListNumber` used before.